src/sync/secret-scanning-alerts.ts

```diff
@@ -0,0 +1,113 @@
+	const smartCursor = new PageSizeAwareCounterCursor(cursor).scale(perPage);
+
+	const { data: secretScanningAlerts  } = await gitHubClient.getSecretScanningAlerts(repository.owner.login, repository.name, {
+		per_page: smartCursor.perPage,
+		page: smartCursor.pageNo,
+		sort: "created",
+		direction: SortDirection.DES
+	});
+
+	if (!secretScanningAlerts?.length) {
+		logger.info({ processingTime: Date.now() - startTime, jiraPayloadLength: 0 }, "Backfill task complete");
+		return {
+			edges: [],
+			jiraPayload: undefined
+		};
+	}
+
+	if (areAllBuildsEarlierThanFromDate(secretScanningAlerts, fromDate)) {
+		logger.info({ processingTime: Date.now() - startTime, jiraPayloadLength: 0 }, "Backfill task complete");
+		return {
+			edges: [],
+			jiraPayload: undefined
+		};
+	}
+	logger.info(`Found ${secretScanningAlerts.length} secret scanning alerts`);
+	const nextPageCursorStr = smartCursor.copyWithPageNo(smartCursor.pageNo + 1).serialise();
+	const edgesWithCursor = [{ secretScanningAlerts, cursor: nextPageCursorStr }];
+
+	const jiraPayload = await transformSecretScanningAlert(secretScanningAlerts, repository, jiraHost, logger,  messagePayload.gitHubAppConfig?.gitHubAppId);
+
+	logger.info({ processingTime: Date.now() - startTime, jiraPayloadLength: jiraPayload?.vulnerabilities?.length }, "Backfill task complete");
+	return {
+		edges: edgesWithCursor,
+		jiraPayload
+	};
+};
+
+const areAllBuildsEarlierThanFromDate = (alerts: SecretScanningAlertResponseItem[], fromDate: Date | undefined): boolean => {
+
+	if (!fromDate) return false;
+
+	return alerts.every(alert => {
+		const createdAt = new Date(alert.created_at);
+		return createdAt.getTime() < fromDate.getTime();
+	});
+
+};
+
+
+const transformSecretScanningAlert = async (
+	alerts: SecretScanningAlertResponseItem[],
+	repository: Repository,
+	jiraHost: string,
+	logger: Logger,
+	gitHubAppId: number | undefined
+): Promise<JiraVulnerabilityBulkSubmitData> => {
+
+	const gitHubClientConfig = await getGitHubClientConfigFromAppId(gitHubAppId, jiraHost);
+
+	const vulnerabilities = alerts.map((alert) => {
+		return {
+			schemaVersion: "1.0",
+			id: `d-${transformRepositoryId(repository.id, gitHubClientConfig.baseUrl)}-${alert.number}`,
+			updateSequenceNumber: Date.now(),
+			containerId: transformRepositoryId(repository.id, gitHubClientConfig.baseUrl),
+			displayName: alert.secret_type_display_name || `${alert.secret_type} secret exposed`,
+			description: "Secret scanning alert",
+			url: alert.html_url,
+			type: "sast",
+			introducedDate: alert.created_at,
+			lastUpdated: alert?.resolved_at || alert.created_at,
+			severity: {
+				level: JiraVulnerabilitySeverityEnum.CRITICAL
+			},
+			identifiers: [{
+				displayName: alert.secret_type,
+				url: alert.html_url
+			}],
+			status: transformGitHubStateToJiraStatus(alert.state, logger),
+			additionalInfo: {
+				content: alert.secret_type
```

Comment:
We might as well add the `alert.html_url` here to the `url` property so that we can link back to this alert in the vulnerability table. We should also add that here as well https://github.com/atlassian/github-for-jira/blob/main/src/transforms/transform-secret-scanning-alert.ts#L35

Reply:
Let's actually just remove `additionalInfo` entirely from all secret scanning alerts since we have this info as the identifier